src/common/SmartUrlInput.js

```diff
@@ -54,53 +54,57 @@ type Props = $ReadOnly<{|
   enablesReturnKeyAutomatically: boolean,
 |}>;
 
-type State = {|
-  /**
-   * The actual input string, exactly as entered by the user,
-   * without modifications by autocomplete.
-   */
-  value: string,
-|};
-
-export default class SmartUrlInput extends PureComponent<Props, State> {
-  static contextType = ThemeContext;
-  context: ThemeData;
-  state = {
-    value: '',
-  };
+export default function SmartUrlInput(props: Props): React$Node {
+  const {
+    defaultProtocol,
+    defaultOrganization,
+    defaultDomain,
+    style,
+    onChangeText,
+    onSubmitEditing,
+    enablesReturnKeyAutomatically,
+  } = props;
 
   // We should replace the fixme with
   // `React$ElementRef<typeof TextInput>` when we can. Currently, that
   // would make `.current` be `any(implicit)`, which we don't want;
   // this is probably down to bugs in Flow's special support for React.
-  textInputRef = React.createRef<$FlowFixMe>();
+  const textInputRef = useRef<$FlowFixMe>();
 
-  unsubscribeFocusListener: () => void;
+  /**
+   * The actual input string, exactly as entered by the user,
+   * without modifications by autocomplete.
+   */
+  const [value, setValue] = useState<string>('');
 
-  componentDidMount() {
-    this.unsubscribeFocusListener = this.props.navigation.addListener('focus', () => {
-      if (this.textInputRef.current) {
+  const themeContext = useContext(ThemeContext);
+
+  // When the route is focused in the navigation, focus the input.
+  // Otherwise, if you go back to this screen from the auth screen, the
+  // input won't be focused.
+  useFocusEffect(
+    useCallback(() => {
+      if (textInputRef.current) {
         // `.current` is not type-checked; see definition.
-        this.textInputRef.current.focus();
+        textInputRef.current.focus();
       }
-    });
-  }
-
-  componentWillUnmount() {
-    if (this.unsubscribeFocusListener) {
-      this.unsubscribeFocusListener();
-    }
-  }
-
-  handleChange = (value: string) => {
-    this.setState({ value });
+    }, []),
+  );
```

Comment:
I didn't find the doc super clear about exactly how `useFocusEffect` behaves, so I looked at the implementation. It appears to be in `node_modules/@react-navigation/core/src/useFocusEffect.tsx`.

There's one thing it does which the old code here didn't, but I think has no practical effect for a different reason: if the screen is already focused when the component first mounts, then `useFocusEffect` will call the callback, while the old code relying on just a listener wouldn't.

(The doc does mention this, [lower down](https://reactnavigation.org/docs/use-focus-effect/#how-is-usefocuseffect-different-from-adding-a-listener-for-focus-event).)

Empirically, when I hit the "Add new account" button to open the screen this appears on, the input gets focused anyway (before these changes). I guess that's due to the `autoFocus` prop we pass to the `TextInput`.

I think I wouldn't call this commit NFC, because it does take some more investigation (like seeing that `autoFocus` prop) to see that it doesn't end up changing how the app visibly behaves.